editors/code/src/test_explorer/run_or_debug.ts

```diff
@@ -0,0 +1,266 @@
+        void vscode.window.showInformationMessage("The test choose config from launch.json and you alredy set Stdio Redirection option. We respect it but could not analytics the output.");
+    } else {
+        const tmpFolderPath = await fs.mkdtemp(path.join(os.tmpdir(), 'ra-test-redirect-'));
+        outputFilePath = path.join(tmpFolderPath, 'output.txt');
+        debugConfig["stdio"] = [null, outputFilePath];
+    }
+
+    runnables.forEach((runnable, index) => {
+        if (runnable.testKind === NodeKind.TestModule) {
+            TestItemControllerHelper.visitTestItemTreePreOrder(testItem => {
+                testRun.enqueued(testItem);
+            }, chosenTestItems[index]!.children);
+        } else {
+            testRun.enqueued(chosenTestItems[index]!);
+        }
+    })
+
+    let debugSession: vscode.DebugSession | undefined;
+    disposables.push(vscode.debug.onDidStartDebugSession((session: vscode.DebugSession) => {
+        // Safe, because concurrently debugging is not allowed.
+        // So the name should not be duplicated
+        if (session.name === debugConfig.name) {
+            debugSession = session;
+        }
+    }));
+
+    const success = await vscode.debug.startDebugging(undefined, debugConfig);
+
+    if (!success || token.isCancellationRequested) {
+        dispose();
+        return;
+    }
+
+    token.onCancellationRequested(async () => {
+        await debugSession?.customRequest('disconnect', { restart: false });
+    });
+
+    return await new Promise<void>((resolve: () => void): void => {
+        disposables.push(
+            vscode.debug.onDidTerminateDebugSession(async (session: vscode.DebugSession) => {
+                if (debugConfig.name === session.name) {
+                    debugSession = undefined;
+                    if (outputFilePath) {
+                        const fileLineContents = (await fs.readFile(outputFilePath, 'utf-8'))
+                            .split(/\r?\n/);
+
+                        chosenTestItems.forEach((chosenTestItem) => {
+                            const outputAnalyzer = new LinesRustOutputAnalyzer(testRun, chosenTestItem);
+                            outputAnalyzer.analyticsLines(fileLineContents);
+                        })
+                    }
+                    dispose();
+                    return resolve();
+                }
+            }),
+        );
+    });
+
+    function dispose() {
+        disposables.forEach(d => d.dispose());
+        disposables.length = 0;
+        testRun.end();
+    }
+}
+
+// refer from playwright-vscode
+/**
+ * @param chosenTestItems The chosen ones of test items. The test cases which should be run should be the children of them.
+ */
+async function runChosenTestItems(testRun: vscode.TestRun, chosenTestItems: readonly vscode.TestItem[], token: vscode.CancellationToken) {
+    const childProcesses: ChildProcessWithoutNullStreams[] = []
+    chosenTestItems.forEach(async (chosenTestItem) => {
```

Comment:
The idea is good, in this way we have multi process, another choice to compute a complex command and keep still one process(similar with your debug parting change). I am not certain which way we should do.

The main issue for multi process is the output would be duplicated, but it's more flexible.

I am not sure which way is better, would update here when got update from RA team